week-3/D-array-filter/exercise.js

```diff
@@ -8,7 +8,11 @@
 
 var pairsByIndexRaw = [[0, 3], [1, 2], [2, 1], null, [1], false, "whoops"];
 
-var pairsByIndex; // Complete this statement
+var pairsByIndex = pairsByIndexRaw.filter(function(item) {
+  if (item !== null || item !== boolean || item !== "string") {
```

Comment:
`item !== boolean` - I think you meant to check the `typeof` that item? Same for the "string" check. Remember that `typeof item` will always return a string.

Also, the function that is passed into the array method `filter` has to return a boolean, not the item :)

Good syntax though!